From 55a2536b6cb6e151bfe67fe88677694935f67687 Mon Sep 17 00:00:00 2001
From: Badiuzzaman Iskhandar <badiuzzaman.azzarfan.bin.iskhandar@intel.com>
Date: Tue, 9 Oct 2018 15:31:22 +0800
Subject: [PATCH 0956/1132] REVERTME [IOTG]: drm/i915/: Add GuC v9.29 and HuC
 v1.07 firmware size

In deferred firmware loading, the value of the firmware sizes were not
known during i915 init. This will cause the i915 driver init to fail.
Add the firmware size as workaround to enable huc/guc loading

Change-Id: Id20318a9553e473e768ced036e24c14efd925ae2
Tracked-On:
Signed-off-by: Badiuzzaman Iskhandar <badiuzzaman.azzarfan.bin.iskhandar@intel.com>
---
 drivers/gpu/drm/i915/intel_wopcm.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_wopcm.c b/drivers/gpu/drm/i915/intel_wopcm.c
index 5ff877e..10d1ea1 100644
--- a/drivers/gpu/drm/i915/intel_wopcm.c
+++ b/drivers/gpu/drm/i915/intel_wopcm.c
@@ -63,6 +63,9 @@
 #define GEN9_GUC_FW_RESERVED	(128 * 1024)
 #define GEN9_GUC_WOPCM_OFFSET	(GUC_WOPCM_RESERVED + GEN9_GUC_FW_RESERVED)
 
+#define GEN9_GUC_9_29_SIZE      ((142 * 1024) + 768)
+#define GEN9_HUC_1_07_SIZE      ((150 * 1024) + 576)
+
 /**
  * intel_wopcm_init_early() - Early initialization of the WOPCM.
  * @wopcm: pointer to intel_wopcm.
@@ -155,8 +158,8 @@ static inline int check_hw_restriction(struct drm_i915_private *i915,
 int intel_wopcm_init(struct intel_wopcm *wopcm)
 {
 	struct drm_i915_private *i915 = wopcm_to_i915(wopcm);
-	u32 guc_fw_size = intel_uc_fw_get_upload_size(&i915->guc.fw);
-	u32 huc_fw_size = intel_uc_fw_get_upload_size(&i915->huc.fw);
+	u32 guc_fw_size = GEN9_GUC_9_29_SIZE;
+	u32 huc_fw_size = GEN9_HUC_1_07_SIZE;
 	u32 ctx_rsvd = context_reserved_size(i915);
 	u32 guc_wopcm_base;
 	u32 guc_wopcm_size;
-- 
2.7.4

